libs/wingc/src/jsify.rs

```diff
@@ -710,6 +695,315 @@ impl JSifier {
 	fn jsify_class_member(&self, member: &ClassMember) -> String {
 		format!("{};", self.jsify_symbol(&member.name))
 	}
+
+	fn jsify_resource(
+		&self,
+		env: &SymbolEnv,
+		name: &Symbol,
+		phase: Phase,
+		parent: &Option<Type>,
+		constructor: &Constructor,
+		methods: &[&(Symbol, FunctionDefinition)],
+	) -> String {
+		assert!(phase == Phase::Preflight);
+
+		// Lookup the resource type
+		let resource_type = env.lookup(name, None).unwrap().as_type().unwrap();
+
+		// Get fields to be captured by resource's client
+		let captured_fields = self.get_captures(resource_type);
+
+		// Jsify inflight client
+		let inflight_methods = methods
+			.iter()
+			.filter(|(_, m)| m.signature.flight == Phase::Inflight)
+			.map(|t| *t)
+			.collect::<Vec<_>>();
+		self.jsify_resource_client(env, name, &captured_fields, &inflight_methods, parent);
+
+		// Get all preflight methods to be jsified to the preflight class
+		let preflight_methods = methods
+			.iter()
+			.filter(|(_, m)| m.signature.flight != Phase::Inflight)
+			.map(|t| *t)
+			.collect::<Vec<_>>();
+
+		let toinflight_method = self.jsify_toinflight_method(name, &captured_fields);
+
+		// Jsify class
+		let resource_class = format!(
+			"class {}{} {{\n{}\n{}\n{}\n}}",
+			self.jsify_symbol(name),
+			if let Some(parent) = parent {
+				format!(" extends {}", self.jsify_type(parent))
+			} else {
+				format!(" extends {}", WINGSDK_RESOURCE)
+			},
+			self.jsify_resource_constructor(constructor, parent.is_none()),
```

Comment:
Will this also work if there is no Ctor?

Reply:
Let's add a test for that

Reply:
We currently force a ctor even if it's empty.
When we change that then `constructor` will be an `Option` and I'll fix it accordingly (we need to call `super(scope,id)` in any case).